lib/core-integration/src/Test/Integration/Scenario/API/Network.hs

```diff
@@ -152,3 +157,71 @@ spec = do
                 r <- request @ApiNetworkInformation ctx
                     Link.getNetworkInfo headers Empty
                 verify r expectations
+
+    describe "NETWORK - Cannot query blockchain parameters with \
+             \invalid arguments" $ do
+        let epochNoWrong =
+                T.pack $show $ (+10) $ fromIntegral @Word31 @Int maxBound
+        let matrix = ["earliest", "invalid", epochNoWrong, "-1"]
+        forM_ matrix $ \arg -> it (show arg) $ \ctx -> do
+            let endpoint = ( "GET", "v2/network/parameters/"<>arg )
+            r <- request @ApiNetworkParameters ctx endpoint Default Empty
+            expectResponseCode @IO HTTP.status400 r
+            expectErrorMessage (errMsg400MalformedEpoch $ T.unpack arg) r
+
+    describe "NETWORK - v2/network/parameters - Methods Not Allowed" $ do
+        let matrix = ["POST", "CONNECT", "TRACE", "OPTIONS"]
+        forM_ matrix $ \method -> it (show method) $ \ctx -> do
+            let endpoint = (method, "v2/network/parameters/latest")
+            r <- request @ApiNetworkParameters ctx endpoint Default Empty
+            expectResponseCode @IO HTTP.status405 r
+            expectErrorMessage errMsg405 r
+
+    describe "NETWORK - HTTP headers" $ do
+        forM_ (getHeaderCases HTTP.status200)
+            $ \(title, headers, expectations) -> it title $ \ctx -> do
+                r <- request @ApiNetworkParameters ctx
+                    Link.getNetworkInfo headers Empty
+                verify r expectations
+
+    it "NETWORK - Cannot query blockchain parameters with epoch later \
+       \than current one" $ \ctx -> do
+        r1 <- request @ApiNetworkInformation ctx
+            Link.getNetworkInfo Default Empty
+        let (ApiT (EpochNo currentEpochNo)) =
+                getFromResponse (#nextEpoch . #epochNumber) r1
+
+        let futureEpochNo = T.pack $ show $ currentEpochNo + 10
```

Comment:
That is good. I'd like to suggest to use boundary value, i.e. (the next neighboring epoch number)

```
let futureEpochNo = T.pack $ show $ currentEpochNo + 1
```
As it is (the boundary value) usually something that is prone to discover potential issue.

Reply:
I think `currentEpochNo + 1` could cause race conditions and flaky tests.

If the current epoch were to change just after we fetched it, then `futureEpochNo` would no longer be the future epoch number.

Maybe, `+ 2` and an explanatory comment in that case.

Reply:
I see there are boundary values tested in the suite, but perhaps it could be organized for better clarity:

positive ["0", "latest",current, current-1] -> expecting 200 and a response
negative404 [current+1, maxPossibleEpochNumber] - > expecting 404
negative400 ["-1", maxPossibleEpochNumber + 1] -> expecting 400

Reply:
it is checked here https://github.com/input-output-hk/cardano-wallet/pull/1306/files#diff-c30d095b1b1142f3da80b28597625b7fR169

Reply:
@piotr-iohk feel free to address more things in your review/testing PR for this task:-)